include/work_queue.h

```diff
@@ -6,40 +6,46 @@
 #include <vector>
 #include "types.h"
 
+struct update_batch {
+  node_id_t node_idx;
+  std::vector<node_id_t> upd_vec;
+};
+
 class WorkQueue {
  public:
   class DataNode {
    private:
     // LL next pointer
     DataNode *next = nullptr;
-    node_id_t node_idx = 0;
-    std::vector<node_id_t> data_vec;
+    std::vector<update_batch> batches;
```

Comment:
I really like the idea of making batches a major part of the code instead of something tacked on and optional.